openmc/summary.py

```diff
@@ -0,0 +1,229 @@
+from collections.abc import Iterable
+import re
+import warnings
+
+import numpy as np
+import h5py
+
+import openmc
+import openmc.checkvalue as cv
+from openmc.region import Region
+
+_VERSION_SUMMARY = 6
+
+
+class Summary(object):
+    """Summary of geometry, materials, and tallies used in a simulation.
+
+    Attributes
+    ----------
+    date_and_time : str
+        Date and time when simulation began
+    geometry : openmc.Geometry
+        The geometry reconstructed from the summary file
+    materials : openmc.Materials
+        The materials reconstructed from the summary file
+    nuclides : dict
+        Dictionary whose keys are nuclide names and values are atomic weight
+        ratios.
+    macroscopics : list
+        Names of macroscopic data sets
+    version: tuple of int
+        Version of OpenMC
+
+    """
+
+    def __init__(self, filename):
+        if not filename.endswith(('.h5', '.hdf5')):
+            msg = 'Unable to open "{0}" which is not an HDF5 summary file'
+            raise ValueError(msg)
+
+        self._f = h5py.File(filename, 'r')
+        cv.check_filetype_version(self._f, 'summary', _VERSION_SUMMARY)
+
+        self._geometry = openmc.Geometry()
+
+        self._fast_materials = {}
+        self._fast_surfaces = {}
+        self._fast_cells = {}
+        self._fast_universes = {}
+        self._fast_lattices = {}
+
+        self._materials = openmc.Materials()
+        self._nuclides = {}
+        self._macroscopics = []
+
+        self._read_nuclides()
+        self._read_macroscopics()
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", openmc.IDWarning)
```

Comment:
Why is this ignore done in init? Would recommend moving this inside `_read_geometry` if it's only specific to that method